meta/src/helper.rs

```diff
@@ -0,0 +1,47 @@
+use crate::utils;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use syn::visit::Visit;
+use syn::{Field, Ident, PathSegment};
+
+pub struct HelperVisitor {
+    name: Ident,
+    functions: Vec<TokenStream2>,
+}
+
+impl HelperVisitor {
+    pub fn new(name: &Ident) -> Self {
+        Self {
+            name: name.clone(),
+            functions: Vec::new(),
+        }
+    }
+
+    fn make_function(name: &Ident, segment: &PathSegment) -> TokenStream2 {
+        let concat = format!("set_{}", name);
+        let concat = Ident::new(&concat, name.span());
+        quote! {
+            pub fn #concat(&mut self, mut #name: #segment) -> #segment {
+                use std::mem::swap;
+                swap(&mut self.#name, &mut #name);
+                #name
+            }
+        }
+    }
+
+    pub fn make_block(&self) -> TokenStream2 {
+        utils::make_block(&self.name, &self.functions)
+    }
+}
+
+impl<'a> Visit<'a> for HelperVisitor {
+    fn visit_field(&mut self, field: &'a Field) {
+        if field.attrs.iter().any(utils::is_context) {
+            if let Some(name) = &field.ident {
+                if let Some(segment) = utils::get_type(field) {
+                    self.functions.push(Self::make_function(name, segment));
+                }
+            }
+        }
+    }
```

Comment:
```suggestion
    fn visit_field(&mut self, field: &'a Field) {
        if !field.attrs.iter().any(utils::is_context) {
            return;
        }

        if let (Some(name), Some(segment)) = (&field.ident, utils::get_type(field)) {
            self.functions.push(Self::make_function(name, segment));
        }
    }
```

Reply:
I'm not sure your suggested tuple destructuring makes things any clearer; better to keep unrelated functionality on separate lines.

Reply:
The modified version looks clearer because it favors the never-nesting approach. I like the change that @KrosFire made.

Early return is more verbose. If we had even more nested `if..let` then this change would highlight the issue even more. The code will keep the indentation level while the original nested code will be a bunch of if's nested inside.